### DISCUSSION TITLE
Docker container and dir permissions

### DISCUSSION THREAD

**Discussion by FredsterCode** [kagemomiji/airsonic-advanced#377] [Q&A]
I'm having some trouble with running the docker container. I'm trying to not run things as root and that might be a cause. I get permission issues as the container starts. It can't create the /var/airsonic/ directory. If I manually create the directory then it can't  create a symlink to ffmpeg in /var/airsonic/transcode/ffmpeg 

Tomorrow I'll try running it as root to check that but root is not a preference. 
Aside from that approach has anyone got tips on their working docker setup. Possibly setting id and gid values differently or creating the directories manually and setting permissions to something. I have tried these approaches already per the guidelines and using a docker compose file. 
("You may change the user running the internal process in one of two ways:

Specifying --user when invoking the docker run command, and providing it with one or both in the format uid:gid
Specifying the PUID or PGID environment variables to the container image when invoking the docker run command (-e PUID=uid -e PGID=gid)
")
for running the container, but I may be missing something as I'm no docker wizard. 
Any help or troubleshooting advice appreciated.

**Comment by FredsterCode** [comment]
I tried running docker with sudo and it still didn't work. In the end I needed to change the ID and GID values to 0. 0 is the id and group id for for root. 
So far it starts up. I have not fully tested transcoding different formats etc yet. 

So my current understanding: the internal id needs to be root so that when it reaches out of the container and engages with /var/airsonic/transcode/ it can create it. I'm going to try to understand a little more around how that path is used and possibly try to map it into the .data/ path. Perhaps something like .data/var/airsonic/transcode/ or even .var/airsonic/transcode to retain its original name. If anyone knows of other situations where using a non root user breaks airsonic-advanced in a container in Ubuntu please let me know.

**Reply by bartmans** [reply]
No doesn't need to be root. But the PUID and PGID should be from a user that has the correct rights.

You can set:
environment:
      - PUID=1026 (this is my UID, change for your situation)
      - PGID=100 (same)

**Comment by RetroButton** [comment]
I also have problems running the docker container not as root ( 0 / 0 ).
The user i want to use for the Docker container definitely has sufficient rights on my media folders and docker folders.
PGID and PUID is double checked, permissions also.
But no chance to not run the container as root.
Any ideas here?

**Comment by pjosuah** [comment]
Hi folks, quick feedback on this: I could get it to work by using `user` as environment variable instead of `PUID:PGID`.
